beacon-light-client/plonky2/balance_verification/balances_input_scheduler_lido.ts

```diff
@@ -0,0 +1,155 @@
+import { JsonRpcProvider, ethers } from 'ethers';
+import { getBalancesInput } from './get_balances_input';
+import { CommandLineOptionsBuilder } from '../cmdline';
+import commonConfig from '../common_config.json';
+import { getBeaconApi } from '@dendreth/relay/implementations/beacon-api';
+import { getLidoWithdrawCredentials } from '@dendreth/utils/balance-verification-utils/utils';
+
+(async () => {
+  const options = new CommandLineOptionsBuilder()
+    .withRedisOpts()
+    .option('rpc-url', {
+      describe: 'The RPC URL',
+      type: 'string',
+      default: process.env.SEPOLIA_RPC,
+    })
+    .option('lido-locator-contract-address', {
+      describe: 'The contract address',
+      type: 'string',
+      demandOption: true,
+    })
+    .option('beacon-node', {
+      alias: 'beacon-node',
+      describe: 'The beacon node url',
+      type: 'array',
+      default: commonConfig['beacon-node'],
+      description: 'Sets a custom beacon node url',
+    })
+    .option('network', {
+      describe: 'The network',
+      type: 'string',
+      default: 'sepolia',
+    })
+    .build();
+
+  const provider = new ethers.JsonRpcProvider(options['rpc-url']);
+  const lidoLocatorContractAddress = options['lido-locator-contract-address'];
+
+  let nextRefSlot = await getNextRefSlot(provider, lidoLocatorContractAddress);
+
+  const beaconApi = await getBeaconApi(options['beacon-node']);
+  const eventSource = beaconApi.subscribeForEvents(['head']);
+
+  const LIDO_WITHDRAWAL_CREDENTIALS = await getLidoWithdrawCredentials(
+    options['network'],
+  );
+
+  eventSource.addEventListener('head', async (event: any) => {
+    const headEpoch = BigInt(JSON.parse(event.data).slot) / 32n;
```

Comment:
Isn't what you want the head slot, not the epoch?

Reply:
great catch